tests/syntax/test_basic.py

```diff
@@ -124,54 +127,128 @@ def test_mutate_scaled():
     scene = sampleScene(scenario)
-    scene.show(block=False)
+    scene.show2D(block=False)
     plt.close()
 
 @pytest.mark.graphical
-def test_show_zoom():
+def test_show2D_zoom():
     scenario = compileScenic("""
-        ego = Object
-        Object at 10@20
+        ego = new Object
+        new Object at 10@20
     """)
     scene = sampleScene(scenario)
-    scene.show(zoom=1, block=False)
+    scene.show2D(zoom=1, block=False)
     xmin, xmax, ymin, ymax = plt.axis()
     assert xmin < 0 and xmax > 10
     assert ymin < 0 and ymax > 20
     plt.close()
+
+def test_mode2D():
+    scenario = compileScenic(
+        'region = RectangularRegion(0 @ 0, 0, 100, 100)\n'
+        'p = new Point on region\n'
+        'op = new OrientedPoint on region\n'
+        'ego = new Object on region\n'
+        'test_obj_1 = new Object in p.visibleRegion\n'
+        'test_obj_2 = new Object in op.visibleRegion\n'
+        'test_obj_3 = new Object in ego.visibleRegion\n'
+    , mode2D=True)
+    for _ in range(5):
+        scene, _ = scenario.generate()
+
+        for obj in scene.objects:
+            assert obj.position[2] == 0
+
+def test_mode2D_heading():
+    with pytest.raises(SpecifierError):
+        compileScenic('ego = new Object with heading 40 deg')
+
+    scenario = compileScenic('ego = new Object with heading 40 deg', mode2D=True)
+    scene, _ = scenario.generate()
+
+    assert scene.egoObject.heading == pytest.approx(math.radians(40))
+
+    with pytest.raises(InvalidScenarioError):
+        compileScenic("""
+            class TestClass:
+                heading: 40 deg
+
+            ego = new TestClass
+        """)
+
+    scenario =  compileScenic("""
+                    class TestClass:
+                        heading: 40 deg
+
+                    ego = new TestClass
+                """, mode2D=True)
+    scene, _ = scenario.generate()
+
+    assert scene.egoObject.heading == pytest.approx(math.radians(40))
+
+def test_mode2D_interference():
+    scenario = compileScenic(
+        'region = RectangularRegion(0 @ 0, 0, 100, 100)\n'
+        'p = new Point on region\n'
+        'op = new OrientedPoint on region\n'
+        'ego = new Object on region\n'
+        'test_obj_1 = new Object in p.visibleRegion\n'
+        'test_obj_2 = new Object in op.visibleRegion\n'
+        'test_obj_3 = new Object in ego.visibleRegion\n'
+    , mode2D=True)
+    for _ in range(5):
+        scene, _ = scenario.generate()
+
+        for obj in scene.objects:
+            assert obj.position[2] == 0
+
+    scenario = compileScenic(
+        'region = RectangularRegion(0 @ 0, 0, 100, 100)\n'
+        'p = new Point on region\n'
+        'op = new OrientedPoint on region\n'
+        'ego = new Object on region\n'
+        'test_obj_1 = new Object in p.visibleRegion\n'
+        'test_obj_2 = new Object in op.visibleRegion\n'
+        'test_obj_3 = new Object in ego.visibleRegion\n'
```

Comment:
Let's put this code in a variable so that it's clear the test is compiling the same scenario twice (just with a difference of mode).